cmd/containerd-shim-runhcs-v1/service_internal_podshim_test.go

```diff
@@ -573,15 +574,43 @@ func Test_PodShim_closeIOInternal_2ndTaskID_2ndExecID_Success(t *testing.T) {
 	}
 }
 
-func Test_PodShim_updateInternal_Error(t *testing.T) {
-	s := service{
-		tid:       t.Name(),
-		isSandbox: true,
+func Test_PodShim_updateInternal_Success(t *testing.T) {
+	s, t1, _, _ := setupPodServiceWithFakes(t)
+
+	var limit uint64 = 100
+	resources := &specs.WindowsResources{
+		Memory: &specs.WindowsMemoryResources{
+			Limit: &limit,
+		},
 	}
 
-	resp, err := s.updateInternal(context.TODO(), &task.UpdateTaskRequest{ID: t.Name()})
+	any, err := typeurl.MarshalAny(resources)
+	if err != nil {
+		t.Fatal(err)
+	}
 
-	verifyExpectedError(t, resp, err, errdefs.ErrNotImplemented)
+	resp, err := s.updateInternal(context.TODO(), &task.UpdateTaskRequest{ID: t1.ID(), Resources: any})
+	if err != nil {
+		t.Fatalf("should not have failed with error, got: %v", err)
+	}
+	if resp == nil {
+		t.Fatalf("should have returned an empty resp")
+	}
+}
+
+func Test_PodShim_updateInternal_Error(t *testing.T) {
+	s, t1, _, _ := setupPodServiceWithFakes(t)
+
+	// resources must be of type *WindowsResources or *LinuxResources
+	resources := &specs.Process{}
+	any, err := typeurl.MarshalAny(resources)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = s.updateInternal(context.TODO(), &task.UpdateTaskRequest{ID: t1.ID(), Resources: any})
+	if err == nil {
```

Comment:
I think we should probably also verify here that the error if actually about the incorrect resource types and not some other error.